ch2driver/pytpcc/runtime/loader.py

```diff
@@ -470,9 +719,66 @@ def generateRegion(self, regionkey):
     ## DEF
 
     ## ==============================================
-    ## generateAddress
+    ## generateCustomerFullName
+    ## ==============================================
+    def generateCustomerFullName(self, c_id):
+        """
+            Returns a full name (first name, middle name, and last name)
+        """
+        c_first = rand.astring(constants.MIN_FIRST, constants.MAX_FIRST)
+        c_middle = constants.MIDDLE
+        assert 1 <= c_id and c_id <= constants.CUSTOMERS_PER_DISTRICT
+        if c_id <= 1000:
+            c_last = rand.makeLastName(c_id - 1)
+        else:
+            c_last = rand.makeRandomLastName(constants.CUSTOMERS_PER_DISTRICT)
+        return [ c_first, c_middle, c_last ]
+    ## DEF
+
+    ## ==============================================
+    ## generateCustomerAddresses
+    ## ==============================================
+    def generateCustomerAddresses(self):
+        """
+            Returns customer addresses
+        """
+        c_addresses = [["shipping"] + self.generateStreetAddress()]
+        c_other_addr_type = ["home", "work", "billing"]
+        c_other_addr = np.random.choice(c_other_addr_type, size=np.random.choice([0,1,2,3]), replace=False)
+        for coa in c_other_addr:
+            c_addresses.append([coa] + self.generateStreetAddress())
+
+        return c_addresses
+    ## DEF
+
+    ## ==============================================
+    ## generateCustomerPhones
+    ## ==============================================
+    def generateCustomerPhones(self):
+        """
+            Returns customer phones
+        """
+        c_phones = [["contact"] + [rand.nstring(constants.PHONE, constants.PHONE)]]
+        c_other_phone_type = ["home", "work", "mobile"]
+        c_other_phone = np.random.choice(c_other_phone_type, size=np.random.choice([0,1,2,3]), replace=False)
```

Comment:
Using `size=np.random.randint(0, 4)` would be better here

Reply:
Done